src/test/java/org/ays/auth/tests/RoleCreateTest.java

```diff
@@ -0,0 +1,153 @@
+    public void createRoleWithNullRoleName() {
+
+        RoleCreatePayload role = RoleCreatePayload.generate();
+        role.setName(null);
+
+        Response response = RoleEndpoints.create(role, accessToken);
+        response.then()
+                .spec(AysResponseSpecs.expectBadRequestResponseSpec())
+                .body("subErrors[0].message", equalTo("must not be blank"));
+    }
+
+    @Test(groups = {"Regression"}, dataProvider = "invalidPermissionIds", dataProviderClass = AysDataProvider.class)
+    public void createRoleWithInvalidPermissionIdList(List<String> permissionIds, AysErrorMessage errorMessage,
+                                                      String field, String type) {
+
+        RoleCreatePayload role = RoleCreatePayload.generate();
+        role.setPermissionIds(permissionIds);
+
+        Response response = RoleEndpoints.create(role, accessToken);
+        response.then()
+                .spec(AysResponseSpecs.expectBadRequestResponseSpec())
+                .spec(AysResponseSpecs.subErrorsSpec(errorMessage, field, type));
+    }
+
+    @Test(groups = {"Regression"})
+    public void createRoleWithNullPermissionId() {
+
+        RoleCreatePayload role = RoleCreatePayload.generate();
+        role.setPermissionIds(null);
+
+        Response response = RoleEndpoints.create(role, accessToken);
+        response.then()
+                .spec(AysResponseSpecs.expectBadRequestResponseSpec())
+                .body("subErrors[0].message", equalTo("must not be empty"));
+    }
+
+    @Test(groups = {"Regression"})
+    public void createRoleWithDuplicatePermissionIds() {
+
+        RoleCreatePayload role = RoleCreatePayload.generate();
+        role.setPermissionIds(List.of("17dd50f6-61fe-4a30-a136-d9b80649e7fe", "17dd50f6-61fe-4a30-a136-d9b80649e7fe"));
+        RoleEndpoints.create(role, accessToken);
+        String roleId = RoleDataSource.findLastCreatedRoleIdByInstitutionId(AysConfigurationProperty
+                .TestVolunteerFoundation.ID);
+
+        List<String> permissionIds = PermissionDataSource.getPermissionIdsFromLastCreatedRole(roleId);
+        List<String> expectedPermissionIds = List.of("17dd50f6-61fe-4a30-a136-d9b80649e7fe");
+
+        Assert.assertEquals(permissionIds.size(), 1, "Permission IDs should be unique in the database.");
+        Assert.assertEquals(permissionIds, expectedPermissionIds);
+    }
+
+    @Test(groups = {"Regression"})
+    public void createRoleByUserWithoutRoleCreatePermission() {
+        LoginPayload login = LoginPayload.generateAsTestVolunteerFoundationUser();
+        String token = loginAndGetAccessToken(login);
+
+        RoleCreatePayload role = RoleCreatePayload.generate();
+
+        Response response = RoleEndpoints.create(role, token);
+        response.then()
+                .spec(AysResponseSpecs.expectUnauthorizedResponseSpec())
+                .body("header", equalTo("AUTH ERROR"))
+                .body("isSuccess", equalTo(false));
+
+    }
```

Comment:
Bu test şöyle daha iyi olabilir. role:create yetkisi olmayan yeni bir kullanıcı oluşturup o kullanıcı üzerinden rol oluşturma adımları test edilebilir. Bu sayede mevcut kullanıcının permissionları değiştirildiğinde testimizin hata vermesinin önüne geçmiş oluruz.

Reply:
Ek olarak header ve isSuccess kısmı response spec içerisinde mevcut kaldırabiliriz

Reply:
@cagla88 kullandigim kullanicinin source page'i landing olarak setlenmis, bu metodu değiştirmeyeceğimiz icin sorun olmaz diye düşünüyorum, LoginPayload'daki son metod generateAsTestVolunteerFoundationUser() adında, institution olmadigi müddetçe hata almayız sanirim, ama yanlis yorumluyorsam söyle lütfen, ona gore düzeltme gerekirse yapayım

Reply:
Bu konuşmayı hatırladım şimdi o zaman tam olarak ne için sorduğunu anlamamışım sanırım. Bu şekilde de kalabilir teste uygun kullanıcı oluşturup ilerlememiz daha doğru bir yaklaşım olabilir.